# Spatial Transcriptomics with Python [Beginnerâ€™s course]

## Sessions:

For each session, a notebook was prepared. You can directly open them on the side or use the links below:

* [Session 2 - Data Preparation and Quality Control](./Session_2_Preprocessing.ipynb)
* [Session 3 - Clustering and Annotation](./Session_3_Clustering_and_Annotation.ipynb)
* [Session 4 - Neighborhood Enrichment Analysis](./Session_4_Neighborhood_Enrichment.ipynb)

You already have your copy of the data, so feel free to change anything you want there. After the course, you can download your scripts or access the original version in the GitHub repository.

All notebooks were created using information from the [scanpy](https://scanpy.readthedocs.io/en/stable/) [1] and [squidpy](https://squidpy.readthedocs.io/en/stable/) [2] documentation.

The dataset is freely available on the [10X website](https://support.10xgenomics.com/spatial-gene-expression/datasets/1.0.0/V1_Human_Lymph_Node).


[1] Wolf *et al.* (2018), Scanpy: large-scale single-cell gene expression data analysis, Genome Biology.
<br>
[2] Pall *et al.* (2022) Squidpy: a scalable framework for spatial omics analysis. Nature Methods.